File: Implemetation, Math/Python/6588 골드바흐의 추측.py
from sys import stdin

def isPrime(number) :
    if number == 1: return False
    if number % 2 == 0 : return False
    for i in range(3, int(number ** 0.5)+1, 2) :
        if number % i == 0 : return False;
    return True

while True :
    n = int(stdin.readline())
    if n == 0 : break
    
    j = n
    for i in range(1, n) :
        j = n-i
        if isPrime(i) and isPrime(j) :
            print("%d = %d + %d" % (i+j, i, j))
            break
    else :
        print("Goldbach's conjecture is wrong.")
    

File: Beakjoon/_etc/C#/6588 골드바흐의 추측.cs
using System;
using static System.Console;

class Program
{
    static bool isPrime(int number) {
        if(number % 2 == 0) return false;
        if(number == 1) return false;
        for(int i = 3; i <= (int)Math.Sqrt(number); i+=2) {
            if(number % i == 0) return false;
        }
        return true;
    }
    static void Main() {
        while(true) {
            string s = ReadLine();
            if(s[0] == '0') break;
            
            int n = int.Parse(s);
            int j = n;
            for(int i = 1; i <= n; i++) {
                j = n-i;
                if(isPrime(i) && isPrime(j)) {
                    WriteLine(String.Format("{0} = {1} + {2}", i+j, i, j));
                    break;
                }
            }
            if(j == 0) {
                WriteLine("Goldbach's conjecture is wrong.");
            }
        }
    }
}
